language-server/src/features/codeAction/extractSubschema.test.ts

```diff
@@ -0,0 +1,217 @@
+                range: {
+                  end: {
+                    character: 5,
+                    line: 7
+                  },
+                  start: {
+                    character: 13,
+                    line: 4
+                  }
+                }
+              },
+              {
+                newText: `,
+  "$defs": {
+    "def1": {
+      "type": "string",
+      "format": "email"
+    }
+  }`,
+                range: {
+                  end: {
+                    character: 3,
+                    line: 8
+                  },
+                  start: {
+                    character: 3,
+                    line: 8
+                  }
+                }
+              }
+            ],
+            textDocument: {
+              uri: documentUri,
+              version: null
+            }
+          }
+        ]
+      },
+      kind: "refactor.extract",
+      title: "Extract 'def1' to $defs"
+    };
+    expect(codeActions?.[0]).toMatchObject(expectedCodeAction);
+  });
+
+  test("Handling existing defs with incremental names def1,def2", async () => {
+    await client.writeDocument("subject.schema.json", `{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "foo": { "$ref": "#/definitions/foo" },
+    "bar": {
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "definitions": {
+    "def1": {
+      "type": "string"
+    }
+  }
+
+}`);
+    documentUri = await client.openDocument("subject.schema.json");
+
+    const codeActions = await client.sendRequest(CodeActionRequest.type, {
+      textDocument: { uri: documentUri },
+      range: {
+        start: { line: 5, character: 11 },
+        end: { line: 8, character: 5 }
+      },
+      context: { diagnostics: [] }
+    });
+    const expectedCodeAction = {
+      edit: {
+        documentChanges: [
+          {
+            edits: [
+              {
+                newText: `{ "$ref": "#/definitions/def2" }`,
+                range: {
+                  end: {
+                    character: 5,
+                    line: 8
+                  },
+                  start: {
+                    character: 11,
+                    line: 5
+                  }
+                }
+              },
+              {
+                newText: `,
+    "def2": {
+      "type": "integer",
+      "minimum": 0
+    }`,
+                range: {
+                  end: {
+                    character: 5,
+                    line: 13
+                  },
+                  start: {
+                    character: 5,
+                    line: 13
+                  }
+                }
+              }
+            ],
+            textDocument: {
+              uri: documentUri,
+              version: null
+            }
+          }
+        ]
+      },
+      kind: "refactor.extract",
+      title: "Extract 'def2' to definitions"
+    };
+    expect(codeActions?.[0]).toMatchObject(expectedCodeAction);
```

Comment:
Although Jest style matchers are supported by vitest, I've been sticking to chai matchers in this project. I think it's nice to keep that consistency whenever possible.

```suggestion
    expect(codeActions?.[0]).to.eql(expectedCodeAction);
```